testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2460,17 +2460,23 @@ public SetColumnTypeSetup(String sourceColumnType, String sourceValueLiteral, St
             requireNonNull(sourceValueLiteral, "sourceValueLiteral is null");
             requireNonNull(newColumnType, "newColumnType is null");
             requireNonNull(newValueLiteral, "newValueLiteral is null");
+            requireNonNull(tableProperty, "tableProperty is null");
         }
 
         public SetColumnTypeSetup withNewValueLiteral(String newValueLiteral)
         {
             checkState(!unsupportedType);
-            return new SetColumnTypeSetup(sourceColumnType, sourceValueLiteral, newColumnType, newValueLiteral, unsupportedType);
+            return new SetColumnTypeSetup(sourceColumnType, sourceValueLiteral, newColumnType, newValueLiteral, unsupportedType, tableProperty);
+        }
+
+        public SetColumnTypeSetup withTableProperty(String tableProperty)
+        {
+            return new SetColumnTypeSetup(sourceColumnType, sourceValueLiteral, newColumnType, newValueLiteral, unsupportedType, Optional.of("WITH (%s)".formatted(tableProperty)));
```

Comment:
`withTableProperty(foo)` should be equivalent do instnatiating `new SetColumnTypeSetup(.... , foo)`
currently the former wraps with `WITH (..)`

move the SQL formatting logic to where it is used (will add separate comment there)